Configure.cmake

```diff
@@ -36,7 +36,13 @@ function(command_arguments PROPNAME)
   set(${PROPNAME} ${quoted_args} PARENT_SCOPE)
 endfunction()
 
+# Detect the path of cmake executable
 
+find_program(CMAKE_EXE_PATH "cmake")
+
+# Detect if cmake is running on Travis
+string(COMPARE NOTEQUAL "" "$ENV{TRAVIS}" RUNNING_ON_TRAVIS)
```

Comment:
I think we should not hard-code Travis detection in the cmake file. Why do you need to detect if we are configuring the build on Travis?